### PR TITLE
fix a lot of things

### PR DESCRIPTION
**Pull request by whateverusername0**
## About the PR

FUCK NEWSTORE
heretic will never work with it

## Why / Balance

image
i hate you and your newstore

:cl:
- fix: made shop great again.
- fix: heretics work properly.
- tweak: Flesh ascension now permanently transforms you into meatball man.
- fix: Fixed heretic ash path not being fully fire immune.

## Release Notes

- **New Features**
  - Introduced a new sales functionality in the store, allowing for discounts based on specified parameters.
  - Added new event handling for heretic abilities related to ascension mechanics.

- **Changes**
  - Simplified listing management by removing cost modifiers from the store listings.
  - Updated store UI elements to reflect changes in listing data structure.
  - Enhanced localization for store listings to include discount information.
  - Updated the store component to manage listings more effectively with new sales parameters.
  - Removed discount handling from uplink operations, streamlining the process.

- **Bug Fixes**
  - Streamlined event handling for damage modifications in the armor system.
  - Removed outdated discount categories from the uplink catalog, simplifying item pricing.

These updates aim to improve user experience by simplifying store interactions and enhancing gameplay mechanics.

### CODE REVIEW

**Comment by whateverusername0** [comment]
@coderabbitai give me 5 reasons why streamlined generic store system is better than the one with integrated discounts

File: Content.Shared/_Goobstation/Heretic/Heretic.Abilites.cs [hunk 1]
```diff
@@ -92,12 +92,19 @@ public sealed partial class EventHereticCascade : InstantActionEvent { }
...
 public sealed partial class HereticVoidBlastEvent : InstantActionEvent { }
 public sealed partial class HereticVoidBlinkEvent : WorldTargetActionEvent { }
 public sealed partial class HereticVoidPullEvent : InstantActionEvent { }
+
+// ascensions
+[Serializable, NetSerializable, DataDefinition] public sealed partial class HereticAscensionAshEvent : EntityEventArgs { }
+[Serializable, NetSerializable, DataDefinition] public sealed partial class HereticAscensionVoidEvent : EntityEventArgs { }
+[Serializable, NetSerializable, DataDefinition] public sealed partial class HereticAscensionFleshEvent : InstantActionEvent { }
+[Serializable, NetSerializable, DataDefinition] public sealed partial class HereticAscensionLockEvent : InstantActionEvent { }
+[Serializable, NetSerializable, DataDefinition] public sealed partial class HereticAscensionBladeEvent : InstantActionEvent { }
```
[the remark this answers is not in the document]
Comment (whateverusername0) [reply Content.Shared/_Goobstation/Heretic/Heretic.Abilites.cs:109]: todo

**Comment by deltanedas** [comment]
how the fuck did discounts break heretic

**Comment by whateverusername0** [comment]
> how the fuck did discounts break heretic

newstore being uncloneable piece of shitcode fucked up heretic since it relied heavily on it

**Comment by whateverusername0** [comment]
epic test fail

**Comment by whateverusername0** [comment]
@Piras314 merge now

**Review by Piras314 — APPROVED** [review]
tested, approved!

**Pull request merged**